tss-esapi/src/ffi/data_zeroize.rs

```diff
@@ -0,0 +1,372 @@
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_named_fields_type {
+    ($tss_type:ident, $($field_name:ident), *) => {
+        impl FfiDataZeroize for $tss_type {
+            fn ffi_data_zeroize(&mut self) {
+                $(
+                    self.$field_name.zeroize();
+                )*
+            }
+        }
+    };
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_structured_named_fields_type {
+    ($tss_type:ident, $($field_name:ident), *) => {
+        impl FfiDataZeroize for $tss_type {
+            fn ffi_data_zeroize(&mut self) {
+                $(
+                    self.$field_name.ffi_data_zeroize();
+                )*
+            }
+        }
+    };
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_named_field_buffer_type {
+    ($tss_type:ident,$buffer_field_name:ident) => {
+        implement_ffi_data_zeroizer_trait_for_named_fields_type!(
+            $tss_type,
+            size,
+            $buffer_field_name
+        );
+    };
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_named_field_structured_buffer_type {
+    ($tss_type:ident,$buffer_field_name:ident) => {
+        impl FfiDataZeroize for $tss_type {
+            fn ffi_data_zeroize(&mut self) {
+                self.size.zeroize();
+                self.$buffer_field_name.ffi_data_zeroize();
+            }
+        }
+    };
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_buffer_type {
+    ($tss_type:ident) => {
+        implement_ffi_data_zeroizer_trait_for_named_field_buffer_type!($tss_type, buffer);
+    };
+}
+
+macro_rules! implement_ffi_data_zeroizer_trait_for_ticket_type {
+    ($tss_type:ident) => {
+        impl FfiDataZeroize for $tss_type {
+            fn ffi_data_zeroize(&mut self) {
+                self.tag.zeroize();
+                self.hierarchy.zeroize();
+                self.digest.ffi_data_zeroize();
+            }
+        }
+    };
+}
+
+// ////////////////////////////////////////////////////////////////////////////////////
+// Implementation of FfiDataZeroize for TPM2B types
+// ////////////////////////////////////////////////////////////////////////////////////
+// TPM2B_AUTH, TPM2B_NONCE, TPM2B_TIMEOUT == TPM2B_DIGEST
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_DIGEST);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_DATA);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_ECC_PARAMETER);
+implement_ffi_data_zeroizer_trait_for_named_field_buffer_type!(TPM2B_ENCRYPTED_SECRET, secret);
+implement_ffi_data_zeroizer_trait_for_named_field_buffer_type!(TPM2B_ID_OBJECT, credential);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_IV);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_MAX_BUFFER);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_MAX_NV_BUFFER);
+implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_PRIVATE);
+//implement_ffi_data_zeroizer_trait_for_buffer_type!(TPM2B_PUBLIC);
```

Comment:
Needed?

Reply:
> Needed?

No it is not needed it is implemented further down.